cmd/livepeer/starter/starter.go

```diff
@@ -526,16 +526,24 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 		}
 
 	} else {
-		var keystoreDir string
-		if _, err := os.Stat(*cfg.EthKeystorePath); !os.IsNotExist(err) {
-			keystoreDir, _ = filepath.Split(*cfg.EthKeystorePath)
+		var keystoreDir = filepath.Join(*cfg.Datadir, "keystore")
+		if keystoreInfo, err := ParseEthKeystorePath(*cfg.EthKeystorePath); err == nil {
+			if keystoreInfo.path != "" {
+				keystoreDir = keystoreInfo.path
+
+				if (keystoreInfo.address != ethcommon.Address{}) {
+					ethKeystoreAddr := keystoreInfo.address.Hex()
+					ethAcctAddr := ethcommon.HexToAddress(*cfg.EthAcctAddr).Hex()
+
+					if (ethAcctAddr == ethcommon.Address{}.Hex()) || ethKeystoreAddr == ethAcctAddr {
+						*cfg.EthAcctAddr = ethKeystoreAddr
+					} else {
+						glog.Fatal("-ethKeystorePath and -ethAcctAddr were both provided, but ethAcctAddr does not match the address found in keystore")
+					}
+				}
+			}
 		} else {
-			keystoreDir = filepath.Join(*cfg.Datadir, "keystore")
-		}
-
-		if keystoreDir == "" {
-			glog.Errorf("Cannot find keystore directory")
-			return
+			glog.Fatal(fmt.Errorf(err.Error()))
```

Comment:
```suggestion
			glog.Fatal(fmt.Errorf(err.Error()))
			return
```

We can return after this error.

Reply:
Added return